App keeps crashing

### Steps to reproduce

1. Download 0.7.0 arm64 release
2. Try to open the app
3. App crashes

### Expected behavior

App should open

### Actual behavior

App crashes

### LibreTube version

0.7.0

### Android version

Android 13

### Other details

```
type: crash
osVersion: google/bluejay/bluejay:13/TP1A.221005.003/2022102800:user/release-keys
package: com.github.libretube:21
process: com.github.libretube

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.github.libretube/com.github.libretube.ui.activities.MainActivity}: java.lang.NullPointerException: bottomNav.menu.findItem(it.id) must not be null
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3677)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3814)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2309)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7904)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
Caused by: java.lang.NullPointerException: bottomNav.menu.findItem(it.id) must not be null
	at com.github.libretube.ui.activities.MainActivity.onCreate(MainActivity.kt:464)
	at android.app.Activity.performCreate(Activity.java:8341)
	at android.app.Activity.performCreate(Activity.java:8320)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1389)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3658)
	... 12 more

```

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/constants/PreferenceKeys.kt
+++ b/app/src/main/java/com/github/libretube/constants/PreferenceKeys.kt
@@ -20,7 +20,7 @@ object PreferenceKeys {
     const val BREAK_REMINDER_TOGGLE = "break_reminder_toggle"
     const val BREAK_REMINDER = "break_reminder"
     const val SAVE_FEED = "save_feed"
-    const val NAVBAR_ITEMS = "nav_bar_items"
+    const val NAVBAR_ITEMS = "navbar_items"
 
     /**
      * Appearance

--- a/app/src/main/java/com/github/libretube/obj/NavBarItem.kt
+++ b/app/src/main/java/com/github/libretube/obj/NavBarItem.kt
@@ -1,7 +1,0 @@
-package com.github.libretube.obj
-
-data class NavBarItem(
-    val id: Int = 0,
-    val title: String = "",
-    var isEnabled: Boolean = true
-)

--- a/app/src/main/java/com/github/libretube/ui/activities/MainActivity.kt
+++ b/app/src/main/java/com/github/libretube/ui/activities/MainActivity.kt
@@ -96,7 +96,7 @@ class MainActivity : BaseActivity() {
         window.navigationBarColor = color
 
         // save start tab fragment id and apply navbar style
-        startFragmentId =  try {
+        startFragmentId = try {
             NavBarHelper.applyNavBarStyle(binding.bottomNav)
         } catch (e: Exception) {
             R.id.homeFragment

--- a/app/src/main/java/com/github/libretube/ui/activities/MainActivity.kt
+++ b/app/src/main/java/com/github/libretube/ui/activities/MainActivity.kt
@@ -95,8 +95,12 @@ class MainActivity : BaseActivity() {
         // sets the navigation bar color to the previously calculated color
         window.navigationBarColor = color
 
-        // save start tab fragment id
-        startFragmentId = NavBarHelper.applyNavBarStyle(binding.bottomNav)
+        // save start tab fragment id and apply navbar style
+        startFragmentId =  try {
+            NavBarHelper.applyNavBarStyle(binding.bottomNav)
+        } catch (e: Exception) {
+            R.id.homeFragment
+        }
 
         // set default tab as start fragment
         navController.graph.setStartDestination(startFragmentId)

--- a/app/src/main/java/com/github/libretube/ui/adapters/NavBarOptionsAdapter.kt
+++ b/app/src/main/java/com/github/libretube/ui/adapters/NavBarOptionsAdapter.kt
@@ -1,16 +1,16 @@
 package com.github.libretube.ui.adapters
 
 import android.view.LayoutInflater
+import android.view.MenuItem
 import android.view.ViewGroup
 import android.widget.Toast
 import androidx.recyclerview.widget.RecyclerView
 import com.github.libretube.R
 import com.github.libretube.databinding.NavOptionsItemBinding
-import com.github.libretube.obj.NavBarItem
 import com.github.libretube.ui.viewholders.NavBarOptionsViewHolder
 
 class NavBarOptionsAdapter(
-    val items: MutableList<NavBarItem>
+    val items: MutableList<MenuItem>
 ) : RecyclerView.Adapter<NavBarOptionsViewHolder>() {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NavBarOptionsViewHolder {
@@ -30,9 +30,9 @@ class NavBarOptionsAdapter(
         val item = items[position]
         holder.binding.apply {
             title.text = item.title
-            checkbox.isChecked = item.isEnabled
+            checkbox.isChecked = item.isVisible
             checkbox.setOnClickListener {
-                if (!checkbox.isChecked && getEnabledItemsCount() < 2) {
+                if (!checkbox.isChecked && getVisibleItemsCount() < 2) {
                     checkbox.isChecked = true
                     Toast.makeText(
                         root.context,
@@ -41,12 +41,12 @@ class NavBarOptionsAdapter(
                     ).show()
                     return@setOnClickListener
                 }
-                items[position].isEnabled = checkbox.isChecked
+                items[position].isVisible = checkbox.isChecked
             }
         }
     }
 
-    private fun getEnabledItemsCount(): Int {
-        return items.filter { it.isEnabled }.size
+    private fun getVisibleItemsCount(): Int {
+        return items.filter { it.isVisible }.size
     }
 }

--- a/app/src/main/java/com/github/libretube/ui/dialogs/NavBarOptionsDialog.kt
+++ b/app/src/main/java/com/github/libretube/ui/dialogs/NavBarOptionsDialog.kt
@@ -62,7 +62,7 @@ class NavBarOptionsDialog : DialogFragment() {
             .setTitle(R.string.navigation_bar)
             .setView(binding.root)
             .setPositiveButton(R.string.okay) { _, _ ->
-                NavBarHelper.setNavBarItems(adapter.items)
+                NavBarHelper.setNavBarItems(adapter.items, requireContext())
                 RequireRestartDialog()
                     .show(requireParentFragment().childFragmentManager, null)
             }

--- a/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
+++ b/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
@@ -255,6 +255,7 @@ class PlayerFragment : BaseFragment(), OnlinePlayerOptions {
                     mainActivity.binding.mainMotionLayout
                 mainMotionLayout.progress = abs(progress)
                 exoPlayerView.hideController()
+                exoPlayerView.useController = false
                 eId = endId
                 sId = startId
             }

--- a/app/src/main/java/com/github/libretube/util/NavBarHelper.kt
+++ b/app/src/main/java/com/github/libretube/util/NavBarHelper.kt
@@ -1,55 +1,70 @@
 package com.github.libretube.util
 
 import android.content.Context
+import android.util.Log
 import android.view.Menu
 import android.view.MenuInflater
 import android.view.MenuItem
 import android.widget.PopupMenu
 import androidx.core.view.forEach
-import com.fasterxml.jackson.core.type.TypeReference
-import com.fasterxml.jackson.databind.ObjectMapper
+import androidx.core.view.get
+import androidx.core.view.size
 import com.github.libretube.R
 import com.github.libretube.constants.PreferenceKeys
-import com.github.libretube.obj.NavBarItem
 import com.google.android.material.bottomnavigation.BottomNavigationView
 import com.google.android.material.navigation.NavigationBarView
 
 object NavBarHelper {
-    private const val preferenceKey = "nav_bar_items"
+    private const val SEPARATOR = ","
 
-    private val mapper = ObjectMapper()
-
-    fun getNavBarItems(context: Context): List<NavBarItem> {
-        return try {
-            val type = object : TypeReference<List<NavBarItem>>() {}
-            mapper.readValue(
-                PreferenceHelper.getString(
-                    preferenceKey,
-                    ""
-                ),
-                type
-            )
+    // contains "-" -> invisible menu item, else -> visible menu item
+    fun getNavBarItems(context: Context): List<MenuItem> {
+        val prefItems = try {
+            PreferenceHelper.getString(
+                PreferenceKeys.NAVBAR_ITEMS,
+                ""
+            ).split(SEPARATOR)
         } catch (e: Exception) {
-            val p = PopupMenu(context, null)
-            MenuInflater(context).inflate(R.menu.bottom_menu, p.menu)
-            val defaultNavBarItems = mutableListOf<NavBarItem>()
-            p.menu.forEach {
-                defaultNavBarItems.add(
-                    NavBarItem(
-                        it.itemId,
-                        it.title.toString(),
-                        it.isEnabled
-                    )
+            Log.e("fail to parse nav items", e.toString())
+            return getDefaultNavBarItems(context)
+        }
+        val p = PopupMenu(context, null)
+        MenuInflater(context).inflate(R.menu.bottom_menu, p.menu)
+
+        if (prefItems.size == p.menu.size) {
+            val navBarItems = mutableListOf<MenuItem>()
+            prefItems.forEach {
+                navBarItems.add(
+                    p.menu[it.replace("-", "").toInt()].apply {
+                        this.isVisible = !it.contains("-")
+                    }
                 )
             }
-            return defaultNavBarItems
+            return navBarItems
         }
+        return getDefaultNavBarItems(context)
     }
 
-    fun setNavBarItems(items: List<NavBarItem>) {
+    private fun getDefaultNavBarItems(context: Context): List<MenuItem> {
+        val p = PopupMenu(context, null)
+        MenuInflater(context).inflate(R.menu.bottom_menu, p.menu)
+        val navBarItems = mutableListOf<MenuItem>()
+        p.menu.forEach {
+            navBarItems.add(it)
+        }
+        return navBarItems
+    }
+
+    fun setNavBarItems(items: List<MenuItem>, context: Context) {
+        val prefString = mutableListOf<String>()
+        val defaultNavBarItems = getDefaultNavBarItems(context)
+        items.forEach { newItem ->
+            val index = defaultNavBarItems.indexOfFirst { newItem.itemId == it.itemId }
+            prefString.add(if (newItem.isVisible) index.toString() else "-$index")
+        }
         PreferenceHelper.putString(
-            preferenceKey,
-            mapper.writeValueAsString(items)
+            PreferenceKeys.NAVBAR_ITEMS,
+            prefString.joinToString(SEPARATOR)
         )
     }
 
@@ -74,14 +89,14 @@ object NavBarHelper {
         // remove the old items
         navBarItems.forEach {
             menuItems.add(
-                bottomNav.menu.findItem(it.id)
+                bottomNav.menu.findItem(it.itemId)
             )
-            bottomNav.menu.removeItem(it.id)
+            bottomNav.menu.removeItem(it.itemId)
         }
 
         navBarItems.forEach { navBarItem ->
-            if (navBarItem.isEnabled) {
-                val menuItem = menuItems.filter { it.itemId == navBarItem.id }[0]
+            if (navBarItem.isVisible) {
+                val menuItem = menuItems.first { it.itemId == navBarItem.itemId }
 
                 bottomNav.menu.add(
                     menuItem.groupId,
@@ -91,6 +106,6 @@ object NavBarHelper {
                 ).icon = menuItem.icon
             }
         }
-        return navBarItems.filter { it.isEnabled }[0].id
+        return navBarItems.first { it.isVisible }.itemId
     }
 }

--- a/app/src/main/res/menu/bottom_menu.xml
+++ b/app/src/main/res/menu/bottom_menu.xml
@@ -18,13 +18,13 @@
 
     <item
         android:id="@+id/downloadsFragment"
-        android:enabled="false"
+        android:visible="false"
         android:icon="@drawable/ic_download_filled"
         android:title="@string/downloads" />
 
     <item
         android:id="@+id/watchHistoryFragment"
-        android:enabled="false"
+        android:visible="false"
         android:icon="@drawable/ic_history_filled"
         android:title="@string/history" />
 

--- a/app/src/main/res/xml/appearance_settings.xml
+++ b/app/src/main/res/xml/appearance_settings.xml
@@ -40,9 +40,8 @@
 
         <Preference
             android:icon="@drawable/ic_home"
-            app:key="nav_bar_items"
+            app:key="navbar_items"
             app:title="@string/navigation_bar" />
-
 
     </PreferenceCategory>
 

